clash-lib/src/Clash/Driver.hs

```diff
@@ -225,14 +225,23 @@ generateHDL
   -> (PrimStep, PrimUnwind)
   -- ^ Hardcoded evaluator (delta-reduction)
   -> [TopEntityT]
-  -- ^ Topentities and associated testbench
+  -- ^ All topentities and associated testbench
+  -> Maybe TopEntityT
+  -- ^ Top entity top compile. If Nothing, all top entities in previous argument
+  -- will be compiled.
   -> ClashOpts
   -- ^ Debug information level for the normalization process
   -> (Clock.UTCTime,Clock.UTCTime)
   -> IO ()
 generateHDL reprs bindingsMap hdlState primMap tcm tupTcm typeTrans eval
-  topEntities0 opts (startTime,prepTime) =
-    go prepTime HashMap.empty (sortTop bindingsMap topEntities2)
+  topEntities0 mainTopEntity opts (startTime,prepTime) =
+    go
+      prepTime
+      HashMap.empty
+      -- TODO: When given a main top entity (by -main-is) we don't currently
```

Comment:
This TODO is for this PR? or a next?

Reply:
I want the current behavior for `clash-shake`, so it's enough for me for now. Unless you think this is behavior is ridiculous, then I'll implement the TODO for this PR.

Reply:
As long as we don't release a version with the TODO still in, i.e. add an issue to the 1.4 milestone to implement the TODO.

Reply:
In that case I'll add it to this PR. I'll request a review when it's done.

Reply:
I've removed support for compiling singular top entities altogether. The TODO is now implemented.